src/main/java/net/sf/jabref/model/database/EntrySorter.java

```diff
@@ -90,8 +90,10 @@ public void databaseChanged(DatabaseChangeEvent e) {
             int pos;
             switch (e.getType()) {
             case ADDED_ENTRY:
-                pos = -Collections.binarySearch(set, e.getEntry(), comp) - 1;
-                set.add(pos, e.getEntry());
+                pos = Collections.binarySearch(set, e.getEntry(), comp);
```

Comment:
Ahh, I was a bit wrong here. The old code didn't work and the current code at least doesn't throw IndexOutOfBoundExceptions but doesn't behave as the old code did (when it worked)...